src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuerySink.java

```diff
@@ -73,4 +76,82 @@ public void clickOnPreviewDataForBigQuerySink() {
+    CdfBigQueryPropertiesActions.enterRangeStart(PluginPropertyUtils.pluginProp(rangeStart));
+  }
+
+  @Then("Enter BigQuery sink property range end {string}")
+  public void enterBigQuerySinkPropertyRangeEnd(String rangeEnd) {
+    CdfBigQueryPropertiesActions.enterRangeEnd(PluginPropertyUtils.pluginProp(rangeEnd));
+  }
+
+  @Then("Enter BigQuery sink property range interval {string}")
+  public void enterBigQuerySinkPropertyRangeInterval(String rangeInterval) {
+    CdfBigQueryPropertiesActions.enterRangeInterval(PluginPropertyUtils.pluginProp(rangeInterval));
+  }
+
+  @Then("Toggle BigQuery sink property require partition filter to true")
+  public void toggleBigQuerySinkPropertyRequirePartitionFilterToTrue() {
+    CdfBigQueryPropertiesActions.toggleRequirePartitionFilter();
+  }
+
+  @Then("Enter BigQuery sink property GCS upload request chunk size {string}")
+  public void enterBigQuerySinkPropertyGCSUploadRequestChunkSize(String chunkSize) {
+    CdfBigQueryPropertiesActions.enterChunkSize(PluginPropertyUtils.pluginProp(chunkSize));
+  }
+
+    @Then("Verify the validation error message for invalid chunk size")
+    public void verifyTheValidationErrorMessageForInvalidChunkSize() {
+      CdfStudioActions.clickValidateButton();
+      String expectedErrorMessage = PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_BQ_INCORRECT_CHUNKSIZE);
+      String actualErrorMessage = PluginPropertyUtils.findPropertyErrorElement("gcsChunkSize").getText();
+      Assert.assertEquals(expectedErrorMessage, actualErrorMessage);
+      String actualColor = PluginPropertyUtils.getErrorColor(PluginPropertyUtils.
+              findPropertyErrorElement("gcsChunkSize"));
+      String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+      Assert.assertEquals(expectedColor, actualColor);
+    }
+
+  @Then("Enter the BigQuery sink properties with incorrect Dataset {string}")
+  public void enterTheBigQuerySinkPropertiesWithIncorrectPropertyValue(String dataset) {
+    CdfBigQueryPropertiesActions.enterBigQueryDataset(PluginPropertyUtils.pluginProp(dataset));
+  }
+
+  @Then("Verify the validation error message for invalid dataset")
+  public void verifyTheValidationErrorMessageForInvalidDataset() {
+    CdfStudioActions.clickValidateButton();
+    String expectedErrorMessage = PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_BQ_INCORRECT_DATASET);
+    String actualErrorMessage = PluginPropertyUtils.findPropertyErrorElement("dataset").getText();
+    Assert.assertEquals(expectedErrorMessage, actualErrorMessage);
+    String actualColor = PluginPropertyUtils.getErrorColor(PluginPropertyUtils.findPropertyErrorElement("dataset"));
+    String expectedColor = ConstantsUtil.ERROR_MSG_COLOR;
+    Assert.assertEquals(expectedColor, actualColor);
+  }
+
+  @Then("Enter the BigQuery sink properties with incorrect table {string}")
+  public void enterTheBigQuerySinkPropertiesWithIncorrectTable(String table) {
```

Comment:
Remove.
Add common definition in BigQueryBase - Enter BigQuery property table {string}